pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2781,18 +2797,21 @@ public void readStartupMessages() throws IOException, SQLException {
   public void receiveParameterStatus() throws IOException, SQLException {
     // ParameterStatus
     pgStream.receiveInteger4(); // MESSAGE SIZE
-    String name = pgStream.receiveString();
-    String value = pgStream.receiveString();
+    final String name = pgStream.receiveCanonicalString();
+    final String value = pgStream.receiveCanonicalString();
```

Comment:
Do we want to do this for the parameter values as well or should we leave those to be read non-canonicalized?

I don't think it matters much either way in the common case where there are few parameter updates. If there's a lot of distinct values then we'll be relying on the GC pressure to eventually clean things up right? There's also the possibility of potentially sensitive user defined GUC values being permanently cached.

Thoughts?

Reply:
I do not have a strong opinion here. Each time a connection is created there is a list of parameters read/processed. Most of those values will repeat for each connection, and several are just "on" or "off".
But if there is legitimate concern about sensitive data being processed through here, I do not believe the gains here are that huge.